src/lib.rs

```diff
@@ -174,29 +174,25 @@ impl<'a> Chart<'a> {
         }
     }
 
-    /// Prints canvas content.
-    pub fn display(&mut self) {
+    pub fn to_string(&mut self) -> String {
         self.figures();
         self.axis();
 
-        let frame = self.canvas.frame();
-        let rows = frame.split('\n').count();
-        for (i, row) in frame.split('\n').enumerate() {
-            if i == 0 {
-                println!("{0} {1:.1}", row, self.ymax);
-            } else if i == (rows - 1) {
-                println!("{0} {1:.1}", row, self.ymin);
-            } else {
-                println!("{}", row);
-            }
-        }
-
-        println!(
-            "{0: <width$.1}{1:.1}",
+        let mut frame = self.canvas.frame();
+        frame.insert_str(frame.find('\n').unwrap(), &format!(" {0:.1}", self.ymax));
```

Comment:
Not real happy with this `.unwrap()`. At the very least I may change it to a `.expect()`